svc-aggregation/rpc/aggregator.go

```diff
@@ -894,3 +897,85 @@ func (a *Aggregator) SendStartUpData(ctx context.Context, req *aggregatorproto.S
 	}
 	return resp, nil
 }
+
+// GetAggregationService is an rpc handler, it gets invoked during GET on AggregationService API (/redfis/v1/AggregationService/)
+func (a *Aggregator) GetResetActionInfoService(ctx context.Context, req *aggregatorproto.AggregatorRequest) (
+	*aggregatorproto.AggregatorResponse, error) {
+	resp := &aggregatorproto.AggregatorResponse{}
+	// Fill the response header first
+	resp.Header = map[string]string{
+		"Date": time.Now().Format(http.TimeFormat),
+		"Link": "</redfish/v1/SchemaStore/en/AggregationService.json>; rel=describedby",
+	}
+	// Validate the token, if user has Login priielege then proceed.
+	//Else send 401 Unauthorised
+	var oemprivileges []string
+	privileges := []string{common.PrivilegeLogin}
+	authResp := a.connector.Auth(req.SessionToken, privileges, oemprivileges)
+	if authResp.StatusCode != http.StatusOK {
+		generateResponse(authResp, resp)
+		return resp, nil
+	}
+	aggregationServiceResponse, _ := json.Marshal(agresponse.ActionInfo{
+		ID:        "ResetActionInfo",
+		OdataType: "#ActionInfo.v1_2_0.ActionInfo",
+		Name:      "Reset Action Info",
+		Parameters: []agresponse.Parameter{
+			{
+				Name:            "ResetType",
+				Required:        true,
+				DataType:        "string",
+				AllowableValues: []string{"On", "ForceOff", "GracefulShutdown", "GracefulRestart", "ForceRestart", "Nmi", "ForceOn", "PushPowerButton"},
+			},
+			{
+				Name:     "TargetURIs",
```

Comment:
there are other two optional parameters are there, add those also. 
BatchSize and DelayBetweenBatchesInSeconds

Reply:
fixed